docs/tutorials/data_provider.md

```diff
@@ -61,3 +61,169 @@ impl AdditiveIdentity {
     }
 }
 ```
+
+## Caching Data Provider
```

Comment:
suggestion: kinda want this to be a separate page because it's so complicated

though perhaps that can be kept in mind for mdbookifying?

Reply:
Filed https://github.com/unicode-org/icu4x/issues/2929 for mdbook, and we can track this there

Reply:
I wasn't exactly sure the best place to put this: it's a bit low-level/detailed for a tutorial, but it's also too big/off-topic for a docs page.

Reply:
yeah i think in the mdbook world we can make this a subchapter